src/main/java/ro/utcn/sd/boti/stackoverflow/service/StackoverflowService.java

```diff
@@ -57,14 +59,45 @@ public void removeQuestion(int id) {
         return repository.findByText(s);
     }
 
+    @Transactional
     public List<Tag> listTags() { return repositoryFactory.getTagRepository().findAll(); }
 
+    @Transactional
     public List<Question> searchByTag(int id) {
         Tag tag = repositoryFactory.getTagRepository().findById(id).get();
         return tag.getQuestions();
     }
 
+    @Transactional
     public Optional<User> findByUserName(String username) {
         return repositoryFactory.getUserRepository().findByUserName(username);
     }
+
+    @Transactional
+    public Optional<Question> findQuestionById(int id) {
+        return repositoryFactory.getQuestionRepository().findById(id);
+    }
+
+    @Transactional
+    public Answer addAnswer(User user, Question question, String text) {
+        Answer answer = new Answer(user, question, text, new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date()));
```

Comment:
If you change the Answer#time field to be a LocalDateTime, then you won't need to do this formatting anymore. YOu can use the LocalDateTime.now() method to get the current time stamp.